### PR TITLE
DIS-142 Decouple javax.ws exceptions with Service layer

### PR DESCRIPTION
**Pull request by jogura**
* Introduces specific DiscussionService exceptions that will be translated into the proper HTTP responses
* Adds new exception mappers
* Updates method protocols
* Updates tests to expect the new exceptions

@armonr @jimmordino

### CODE REVIEW

**Comment by jimmordino** [comment]
Given this, can we remove the throws clauses from service methods?  Will we always be throwing a subclass of DiscussionServiceException once we refactor to use these exception everywhere?

**Comment by jogura** [comment]
>Given this, can we remove the throws clauses from service methods? Will we always be throwing a subclass of DiscussionServiceException once we refactor to use these exception everywhere?

Already done [here](https://github.com/Wikia/pandora/pull/290/files#diff-5eee0cf281e8ad8835fd79cd49ff0fa5L41). Just needed to add the new exception mappers and clean up missed references. 90d853c

**Comment by jimmordino** [comment]
Yes, I'm asking because I've been adding throws clauses as part of my service refactoring - I need to remove those now.

File: service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java [hunk 1]
```diff
@@ -9,6 +10,7 @@
 @FunctionalInterface
 public interface Authorizable {
 
-  boolean authorize(User user, Integer siteId, Long contentId) throws UnauthorizedException;
+  boolean authorize(User user, Integer siteId, Long contentId)
+      throws ForbiddenException, NotFoundException;
```
Comment (armonr) [inline service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java]: So can this just be `throws DiscussionServiceException`?

**Comment by armonr** [comment]
LGTM

File: service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java (hunk 1 above)
Comment (jogura) [inline service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java]: I chose to be explicit.

**Comment by jogura** [comment]
The test failures are outside of the discussion service.

File: service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java (hunk 1 above)
Comment (jimmordino) [inline service/discussion/src/main/java/com/wikia/discussionservice/auth/Authorizable.java]: :clap:

**Pull request merged**